include/core/shape.h

```diff
@@ -531,4 +533,138 @@ class CompositeShape : public Shape<dim>
   std::vector<std::shared_ptr<Shape<dim>>> components;
 };
 
+
+// RBF Shapes express the signed distance function as a linear combination of
+// Radial Basis Functions, which have a defined support radius and basis
+// function. A collection of nodes and weights compose the object.
+template <int dim>
+class RBFShape : public Shape<dim>
+{
+public:
+  /**
+   * @param support_radius the scaling of the reach of the nodes
+   * @param basis_function the basis function that was used to parametrize the RBF object
+   * @param weight the weighting associated to each node for the sum operation
+   * @param nodes the center of each basis function
+   */
+  RBFShape<dim>(const std::vector<double>         support_radius,
+                const std::vector<unsigned int>   basis_function,
+                const std::vector<double>         weight,
+                const std::vector<Tensor<1, dim>> nodes,
+                const Point<dim> &                position,
+                const Tensor<1, 3> &              orientation)
+    : Shape<dim>(support_radius[0], position, orientation)
+    , weight(weight)
+    , nodes(nodes)
+    , support_radius(support_radius)
+    , basis_function(basis_function)
+  {
+    size_t number_of_nodes = weight.size();
+
+    Point<dim>     high_bounding_point = Point<dim>();
+    Point<dim>     low_bounding_point  = Point<dim>();
+    Point<dim>     bounding_box_center = Point<dim>();
+    Tensor<1, dim> half_lengths        = Tensor<1, dim>();
+    for (int d = 0; d < dim; d++)
+      {
+        high_bounding_point[d] = std::numeric_limits<double>::lowest();
+        low_bounding_point[d]  = std::numeric_limits<double>::max();
+        for (size_t i = 0; i < number_of_nodes; i++)
+          {
+            if (low_bounding_point[d] > nodes[i][d])
+              low_bounding_point[d] = nodes[i][d];
+            if (high_bounding_point[d] < nodes[i][d])
+              high_bounding_point[d] = nodes[i][d];
+          }
+        bounding_box_center[d] =
+          0.5 * (low_bounding_point[d] + high_bounding_point[d]);
+        half_lengths[d] =
+          0.5 * (high_bounding_point[d] - low_bounding_point[d]);
+      }
+    bounding_box = std::make_shared<Rectangle<dim>>(half_lengths,
+                                                    bounding_box_center,
+                                                    Tensor<1, 3>());
+  }
+
+  double
+  value(const Point<dim> & evaluation_point,
+        const unsigned int component = 0) const override;
+
+  std::shared_ptr<Shape<dim>>
+  static_copy() const override;
+
+  double
+  displaced_volume(const double fluid_density) override;
+
+  double
+  // Inspired by Optimad Bitpit. https://github.com/optimad/bitpit
+  evaluate_basis_function(const unsigned int basis_function_id,
+                          const double       distance) const;
+
+  double
+  wendlandc2(double) const;
```

Comment:
Following the review comments in shape.cc

```suggestion
  wendlandc2(const double) const;
```